### PR TITLE
Set `--use-repodata-zst` argument as `NULL`, rename `no_repodata_zst` to `use_repodata_zst`

### LINKED ISSUE #13528: Add matching `use_repodata_zst` flag, environment variable to match command line argument.
### Checklist

- [X] I added a descriptive title
- [X] I searched open requests and couldn't find a duplicate

### What is the idea?

The `--no-repodata-zst` flag, used to suppress a default check for `repodata.json.zst`, should have counterparts in `.condarc` and an environment variable.

Also renamed to `--[no-]use-repodata-zst`

### Why is this needed?

_No response_

### What should happen?

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by dholth**
### Description

Use NULL in argparse default value to avoid argparse overriding config file or environment variable.

Fix #13528 

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

**Comment by jezdez** [comment]
pre-commit.ci autofix

File: conda/cli/helpers.py [hunk 1]
```diff
@@ -13,6 +13,50 @@
     _MutuallyExclusiveGroup,
 )
 
+try:
+    from argparse import BooleanOptionalAction
+except:
+    # Python < 3.9
+    from argparse import Action
+    class BooleanOptionalAction(Action):
```
Comment (dholth) [inline conda/cli/helpers.py:21]: Messy to copy/paste for <= py 3.8; Implements click-style `--[no-]-repodata-zst` option pair

File: conda/gateways/repodata/__init__.py [hunk 2]
```diff
@@ -106,7 +106,7 @@ def get_repo_interface() -> type[RepoInterface]:
                 f"Is the required jsonpatch package installed?  {e}"
             )
 
-    if not context.no_repodata_zst:
+    if context.repodata_use_zst:
```
Comment (jezdez) [inline conda/gateways/repodata/__init__.py:109]: Loving this non-double-negative here :D

**Review by jezdez — APPROVED** [review]

**Comment by jezdez** [comment]
pre-commit.ci autofix

**Review by jezdez — APPROVED** [review]

**Pull request merged**